control_panel_api/models.py

```diff
@@ -115,6 +115,18 @@ class Meta:
         ordering = ('id',)
 
 
+class S3BucketQuerySet(models.QuerySet):
+    def accessible_by(self, user):
+        return self.filter(
+            users3buckets__user=user,
+        )
+
+    def accessible_by_admin(self, user):
```

Comment:
Maybe name could be `administered_by()` as "accessible" is a given when you're an admin.

Reply:
One thing to watch out for is that this double application of filter cause a double `JOIN` in the resulting SQL query. e.g. for each of the `UserS3Bucket` records with user/is_admin matching there will be two results.

As far as I can tell we only use this in an if in the `UserS3BucketPermissions` class so this issue should not be user-facing (e.g. users shouldn't see duplicated entries when listing `UserS3Bucket`).

A possible solution could be to write this method as:

```
def administered_by(self, user):
        return self.filter(
            users3buckets__user=user,
            users3buckets__is_admin=True,
        )
```

I know this is a bit of a duplication. Maybe there is a way to tell django to return unique results.

Reply:
👍

Reply:
Oh damn, I thought it would merge them.  I'll go back to original implementation with slight dupe code.